File: studymultimodule/module-api/src/main/java/usada/pekora/ArticleController.java
package usada.pekora;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import usada.pekora.article.CreateArticleUsecase;
import usada.pekora.article.DeleteArticleUsecase;
import usada.pekora.article.GetArticleUsecase;
import usada.pekora.article.model.Article;
import usada.pekora.article.model.dto.CreateArticleRequest;

@RestController
@RequestMapping("/api/articles")
@RequiredArgsConstructor
public class ArticleController {
    private final CreateArticleUsecase createArticleUsecase;
    private final GetArticleUsecase getArticleUsecase;
    private final DeleteArticleUsecase deleteArticleUsecase;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Mono<Void> create(@RequestBody CreateArticleRequest request) {
        return createArticleUsecase.create(request);
    }

    @GetMapping
    public Flux<Article> findAll() {
        return getArticleUsecase.findAll();
    }

    @DeleteMapping("/{articleId}")
    public Mono<Void> delete(@PathVariable String articleId) {
        return deleteArticleUsecase.deleteById(articleId);
    }

}

File: mediumSpringWebMVC/src/main/kotlin/com/example/mediumspringwebmvc/domain/article/api/ArticleController.kt
package com.example.mediumspringwebmvc.domain.article.api

import com.example.mediumspringwebmvc.domain.article.dto.ArticleDetailResponse
import com.example.mediumspringwebmvc.domain.article.dto.ArticleRequest
import com.example.mediumspringwebmvc.domain.article.dto.ArticleResponse
import com.example.mediumspringwebmvc.domain.article.component.*
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/articles")
class ArticleController(
    private val cudArticleComponent: CUDArticleComponent,
    private val readArticleComponent: ReadArticleComponent
) {

    @PostMapping
    fun create(@RequestBody request: ArticleRequest) = cudArticleComponent.create(request)

    @GetMapping
    fun findAll(): List<ArticleResponse> = readArticleComponent.findAll()

    @GetMapping("/{id}")
    fun findById(@PathVariable id: Long): ArticleDetailResponse = readArticleComponent.findById(id)

    @PatchMapping("/{id}")
    fun update(@PathVariable id: Long, @RequestBody request: ArticleRequest) =
        cudArticleComponent.update(id, request)

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long) = cudArticleComponent.delete(id)
}